### DISCUSSION TITLE
lsp symbols: avoid opening picker when `query` results match nothing

### DISCUSSION THREAD

**Discussion by gennaro-tedesco** [ibhagwan/fzf-lua#1630] [Q&A] [answered]
When issueing `require('fzf-lua').lsp_{live|workspace|document}_symbols({query=<query>})` with the additional query parameters for symbols, the picker opens also in case no results are being matched by `query`. Is it possible to check the "return value" of said results before opening the picker window? I actually remember reading on a specific issue where you introduced such return value for the coroutine (but cannot find it anymore, unfortunately).

I have tried using the `fzf` default option `--exit-0`, which does indeed exit the picker, it however raises an exception 
```
[Fzf-lua] unsupported action: 'nil', type:nil
```

**Comment by ibhagwan** [comment, marked as answer]
> When issueing require('fzf-lua').lsp_{live|workspace|document}_symbols({query=<query>}) with the additional query parameters for symbols, the picker opens also in case no results are being matched by query. Is it possible to check the "return value" of said results before opening the picker window? I actually remember reading on a specific issue where you introduced such return value for the coroutine (but cannot find it anymore, unfortunately).

This would only possible when using `async=false` as with async queries we always open the UI and wait for results.

> I have tried using the fzf default option --exit-0, which does indeed exit the picker, it however raises an exception

This is something I can most likely fix.

**Comment by gennaro-tedesco** [comment]
I see, it makes sense indeed, I suppose one should just use `async=false` without having to set the external `fzf` option to `--exit-0`. I confirm that with the async setting the picker does not open if no matching results are found!

**Reply by ibhagwan** [reply]
> I see, it makes sense indeed, I suppose one should just use `async=false` without having to set the external `fzf` option to `--exit-0`. I confirm that with the async setting the picker does not open if no matching results are found!

That would be the better way, in any event `--exit-0` could be useful in other places, should be fixed with https://github.com/ibhagwan/fzf-lua/commit/7022b5b002bb7488e1be4470bad2342bcc8d4d99.
